build-system/tasks/check-types.js

```diff
@@ -236,9 +236,10 @@ async function typeCheck(targetName) {
   // If srcGlobs and externGlobs are defined, determine the externs/extraGlobs
   if (srcGlobs.length || externGlobs.length) {
     opts.externs = externGlobs.flatMap(globby.sync);
-    const externSet = new Set(opts.externs);
+
     // Included globs should explicitly exclude any externs
-    opts.extraGlobs = srcGlobs.flatMap(globby.sync).filter(file => !externSet.has(file));
+    const excludedExterns = externGlobs.map((s) => `!${s}`);
+    opts.extraGlobs = globby.sync(srcGlobs.concat(excludedExterns));
```

Comment:
These lines should be reverted; explained in another comment thread, can walk through offline if that would be helpful

Reply:
Discussing offline and in that thread.